test/forms/AddressFormProviderSpec.scala

```diff
@@ -231,7 +233,37 @@ class AddressFormProviderSpec extends SpecBase with FieldBehaviours with UserAdd
 
     "must not bind for non-BT postcodes when the user is a XI trader" in {
 
-      val expectedResult = Seq(FormError(postcodeField, mustBeBTPostcodeKey, Seq()))
+      val expectedResult = Seq(FormError(postcodeField, mustBeBTPostcodeKey(ConsignorAddressPage), Seq()))
+
+      xiForm.bind(formAnswersMap(Some(postcodeField), Some("B1 1AA"))).errors mustBe expectedResult
+    }
+
+    "must bind for postcodes beginning with 'BT' for a XI trader" in {
+
+      xiForm.bind(formAnswersMap(Some(postcodeField), Some("BT1 1AA"))).errors mustBe empty
+    }
+  }
+
+  "for the ConsigneeAddress Page" - {
+
+    lazy val gbForm = new AddressFormProvider()(ConsigneeAddressPage)(dataRequest(FakeRequest(), emptyUserAnswers, ern = testGreatBritainErn))
+    lazy val xiForm = new AddressFormProvider()(ConsigneeAddressPage)(dataRequest(FakeRequest(), emptyUserAnswers, ern = testNorthernIrelandErn))
```

Comment:
These are testing the consignors ERN not the entered consignee ERN (i.e., on CAM-NEE01)